### PR TITLE
Remove READ support for compatDetails on ContainerRuntime ops

### PR DESCRIPTION
**Pull request by MarioJGMsoft**
#### Description 
Search for compatDetails in the code - all  some usages can now be removed.  Maintaining this behavior has been a pain in tests, and it is not a good feature.

It was only ever used for GC, and wasn't as useful (or safe) as expected.  See discussion in ADO: https://dev.azure.com/fluidframework/internal/_workitems/edit/17024 - RC1 clients don't know about TombstoneLoaded subtype

Fixes issue: [AB#17024](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/17024) 

The full removal of this feature is planned for next sprint (10/14 - 11/1)

### CODE REVIEW

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -2978,27 +2979,12 @@ export class ContainerRuntime
...
-							}),
-						},
-					);
-					this.closeFn(error);
-					throw error;
-				}
+				const error = getUnknownMessageTypeError(
+					message.type,
+					"validateAndProcessRuntimeMessage" /* codePath */,
+				);
```
Comment (markfields) [inline packages/runtime/container-runtime/src/containerRuntime.ts:2985]: Here we have the sequenced message to pass as the third parameter, it will get additional details added to the log.

File: packages/runtime/container-runtime/src/test/containerRuntime.spec.ts [hunk 2]
```diff
@@ -1125,208 +1121,6 @@ describe("Runtime", () => {
...
-					existing: false,
-					requestHandler: undefined,
-					runtimeOptions: {
-						enableGroupedBatching: false,
-					},
-					provideEntryPoint: mockProvideEntryPoint,
-				});
-			});
-
-			it("can submit op compat behavior (temporarily still available for GC op)", async () => {
```
Comment (markfields) [inline packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:1143]: Let's leave this one, since we are still doing it for GC (at least for now)

File: packages/runtime/container-runtime/src/test/containerRuntime.spec.ts [hunk 3]
```diff
@@ -1125,208 +1121,6 @@ describe("Runtime", () => {
...
-							envelope.contents,
-							localOpMetadata,
-						);
-					},
-				} satisfies Partial<ChannelCollection>;
-
-				return patched;
-			}
-
-			it("Op with unrecognized type and 'Ignore' compat behavior is ignored by resubmit", async () => {
```
Comment (markfields) [inline packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:1202]: Let's have a test that shows that we actually DO throw/close even if it says "Ignore".  Should be able to largely reuse this test.

File: packages/runtime/container-runtime/src/test/containerRuntime.spec.ts [hunk 4]
```diff
@@ -1125,208 +1121,6 @@ describe("Runtime", () => {
...
-				changeConnectionState(patchedContainerRuntime, true, mockClientId);
-
-				assert.strictEqual(
-					submittedOps.length,
-					3,
-					"Only 3 messages should be sent - Do not resubmit the future/unknown op",
-				);
-			});
-
-			it("Op with unrecognized type and no compat behavior causes resubmit to throw", async () => {
```
Comment (markfields) [inline packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:1232]: This test should stay as well - this behavior shouldn't be changing

Comment (markfields) [reply packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:1232]: Actually just keep tests for the "process" codepath, don't worry about Resubmit or applyStashed ops, those were only needed for such crazy corner cases.

File: packages/runtime/container-runtime/src/test/containerRuntime.spec.ts [hunk 5]
```diff
@@ -1125,208 +1121,6 @@ describe("Runtime", () => {
...
-					contents: JSON.stringify(futureRuntimeMessage),
-					type: MessageType.Operation,
-					sequenceNumber: 123,
-					clientId: "someClientId",
-					minimumSequenceNumber: 0,
-				};
-				containerRuntime.process(packedOp as ISequencedDocumentMessage, false /* local */);
-			});
-
-			it("process remote op with unrecognized type and 'FailToProcess' compat behavior", async () => {
```
Comment (markfields) [inline packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:1279]: Don't need to test this IMO - if "Ignore" compat behavior still results in an error, it's not very interesting to test this special value that doesn't even exist in the codebase anymore.

File: packages/test/test-end-to-end-tests/src/test/batching.spec.ts [hunk 6]
```diff
@@ -157,49 +156,6 @@ describeCompat("Flushing ops", "NoCompat", (getTestObjectProvider, apis) => {
 		await provider.ensureSynchronized();
 	}
 
-	it("[DEPRECATED] can send and a batch containing a future/unknown op type", async () => {
```
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/batching.spec.ts:160]: Leave this for now

**Review by markfields — COMMENTED** [review]
There's one other place to update, which is the "Stress Tests".  Search the code for `futureop`, you'll find a handful of hits under `packages\test\test-service-load\`.  Remove those too - otherwise our OCEs will probably get an incident when the stress tests try to run and these future ops are continuously causing the container to close.

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 1 above)
Comment (MarioJGMsoft) [reply packages/runtime/container-runtime/src/containerRuntime.ts:2985]: This would also apply to the reSubmitCore function, correct?

File: packages/runtime/container-runtime/src/test/containerRuntime.spec.ts [hunk 7]
```diff
@@ -1198,7 +1247,6 @@ describe("Runtime", () => {
 
 				return patched;
 			}
-
```
Comment (markfields) [inline packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:1201]: SUPER nit: add this line back (only if you have to come back and make other changes). Just to avoid unnecessary churn in the diff

Comment (markfields) [reply packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:1201]: Related: Consider moving the two "new" tests above down where the similar tests were below to minimize the diff.

File: packages/test/test-service-load/src/loadTestDataStore.ts [hunk 8]
```diff
@@ -593,11 +592,6 @@ class LoadTestDataStore extends DataObject implements ILoadTest {
 		const opsSendType = config.testConfig.opsSendType ?? "staggeredReadWrite";
 		const opsPerCycle = (config.testConfig.opRatePerMin * cycleMs) / 60000;
 		const opsGapMs = cycleMs / opsPerCycle;
-		const futureOpPeriod =
-			config.testConfig.futureOpRatePerMin === undefined ||
```
Comment (markfields) [inline packages/test/test-service-load/src/loadTestDataStore.ts:597]: Two other files to update related to this:  testConfigFile.ts and testConfig.json

**Review by markfields — COMMENTED** [review]
Couple more files to update under `test-service-load` dir, I left a comment about it. Otherwise looking good!

**Comment by markfields** [comment]
@agarwal-navin -- FYI on this

**Comment by agarwal-navin** [comment]
Please remove `AB#17024` from the PR title. It's not meaningful. You can add it to the description such that is hydrated to a link by the ADO bot.

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 9]
```diff
@@ -236,19 +239,32 @@ import {
...
+ * @param sequencedMessage - The sequenced message that contained the unexpected message type.
+ *
  */
-function compatBehaviorAllowsMessageType(
-	_unknownContainerRuntimeMessageType: UnknownContainerRuntimeMessage["type"],
-	compatBehavior: "Ignore" | "FailToProcess" | undefined,
-): boolean {
-	// undefined defaults to same behavior as "FailToProcess"
-	return compatBehavior === "Ignore";
+function getUnknownMessageTypeError(
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:253]: This function seems unnecessary. It just creates a data processing error and it would be more readable if it was just inline at the call sites.

**Comment by agarwal-navin** [comment]
What is the plan to remove writing it as well? Per my discussion with Mark, it may be okay to remove that too. It will be good to mention in the description when will this feature be completely removed :-).
Other than that, it looks good!

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 9 above)
Comment (markfields) [reply packages/runtime/container-runtime/src/containerRuntime.ts:253]: It gets type safety relative to the switch statement.  If the set of valid values changes, the call to this function will have a compilation break.

File: packages/runtime/container-runtime/src/test/containerRuntime.spec.ts [hunk 10]
```diff
@@ -1220,39 +1219,11 @@ describe("Runtime", () => {
...
-					type: "FUTURE_TYPE" as any,
-					contents: "3",
-					// No compatDetails so it will throw on resubmit.
-				});
-
-				assert.strictEqual(
-					submittedOps.length,
-					0,
-					"no messages should be sent while disconnected",
+				assert.throws(
```
Comment (markfields) [inline packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:1222]: Update title of this test

File: packages/runtime/container-runtime/src/test/containerRuntime.spec.ts [hunk 11]
```diff
@@ -1273,32 +1244,11 @@ describe("Runtime", () => {
...
-					sequenceNumber: 123,
-					clientId: "someClientId",
-					minimumSequenceNumber: 0,
-				};
 				assert.throws(
 					() =>
 						containerRuntime.process(packedOp as ISequencedDocumentMessage, false /* local */),
 					(error: IErrorBase) => error.errorType === ContainerErrorTypes.dataProcessingError,
-					"Ops with unrecognized type and 'FailToProcess' compat behavior should fail to process",
+					"Ops with unrecognized type and no specified compat behavior should fail to process",
```
Comment (markfields) [inline packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:1251]:
```suggestion
					"Ops with unrecognized type and 'Ignore' compat behavior should fail to process",
```

**Comment by MarioJGMsoft** [comment]
> What is the plan to remove writing it as well? Per my discussion with Mark, it may be okay to remove that too. It will be good to mention in the description when will this feature be completely removed :-). Other than that, it looks good!

I'll work with @markfields to look at the data and make sure that it is completely removable. Then we will create a task and I will work on it next sprint.

**Review by markfields — APPROVED** [review]
two typos to fix and then LGTM!

File: packages/runtime/container-runtime/src/test/containerRuntime.spec.ts (hunk 10 above)
Comment (MarioJGMsoft) [reply packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:1222]: The title is appropriate in my opinion since it's: "Op with unrecognized type and 'Ignore' compat behavior is ignored by resubmit" and that's what happens in the test.
